test/functional/rpc_budget.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 The PIVX developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test RPC commands for budget proposal creation, submission, and verification."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+
+class BudgetProposalTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def run_test(self):
+        budgetcycleblocks = 144
+        nextsuperblock = self.nodes[0].getnextsuperblock()
+        address = self.nodes[0].getnewaddress()
+        scheme = 'http://'
+        url = 'test.com'
+        longurl = 'test.com/longer-than-allowed-url-string-should-fail-if-url-is-more-than-64-characters'
+        name = 'testinvalid'
+        longname = 'ThisProposalNameHasMoreThanTwentyCharacters'
+        numcycles = 2
+        cycleamount = 100
+
+        self.log.info("Test with long name")
+        assert_raises_rpc_error(-8, "Invalid proposal name, limit of 20 characters.", self.nodes[0].preparebudget,
+                                longname, scheme + url, numcycles, nextsuperblock, address, cycleamount)
+
+        self.log.info("Test with long URL")
+        assert_raises_rpc_error(-8, "Invalid URL:", self.nodes[0].preparebudget,
```

Comment:
I'm thinking it would be best to test the exact expected result; especially now that I notice the preamble is different between the length error and the scheme error (and thus may end up changed in the future to have them consistent).
`Invalid URL: 92 exceeds limit of 64 characters."` (as the test is written)

As it stands, if the scheme test code changes to "Invalid URL:  check scheme"; the length test will have a false positive.

This also helps point out that the test needs to change in parallel with any changes to the code.

Reply:
fixed